thunder/torch/__init__.py

```diff
@@ -472,31 +472,12 @@ def to(
     copy: bool = False,
     memory_format: None | torch.memory_format = None,
 ) -> TensorLike:
-
-    input_device = a.device
-    input_dtype = a.dtype
-    if copy and not _will_to_return_self(input_device, input_dtype, device, dtype, memory_format, copy):
-        b = prims.empty(a.shape, device=input_device, dtype=input_dtype)
-        return _copy_(b, a)
```

Comment:
empty + copy_ can certainly be replaced with a clone operation without the implications of an in-place operation. Nice change!